promgen/tests/test_models.py

```diff
@@ -34,3 +36,32 @@ def test_validators(self):
             validators.metricname(
                 "[[this.$el.ownerDocument.defaultView.alert(1337)]]",
             )
+
+    @mock.patch("django.dispatch.dispatcher.Signal.send")
+    def test_custom_save_rule(self, mock_post):
+        # Check if annotation["rule"] is automatically set to be {domain}/rule/{id} when creating a new rule
+        rule = models.Rule(
+            name="example-rule",
+            content_type=ContentType.objects.get_for_model(models.Site),
+            object_id=1,
+            clause="up==1",
+            duration="1s"
+        )
+        rule.save()
+        self.assertEqual(resolve_domain("rule-detail", rule.pk), rule.annotations["rule"])
+
+        # Check if annotation["rule"] is automatically set to be {domain}/rule/{id} when updating an existed rule
```

Comment:
```suggestion
        # Check if annotation["rule"] is automatically set to be {domain}/rule/{id} when updating an
        # existed rule
```

Please wrap your lines at 100 characters, including comments.